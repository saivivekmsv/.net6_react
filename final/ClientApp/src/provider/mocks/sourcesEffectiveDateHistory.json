[
  {
    "startDate": "2020-10-21T12:10:45.741",
    "endDate": "2020-10-22T00:00:00",
    "updatedBy": "Admin 123",
    "updatedOn": "2020-10-23T00:00:00"
  },
  {
    "startDate": "2020-10-20T12:10:45.741",
    "endDate": "2020-10-21T00:00:00",
    "updatedBy": "Admin 123",
    "updatedOn": "2020-10-22T00:00:00"
  },
  {
    "startDate": "2020-10-15T12:10:45.741",
    "endDate": "2020-10-16T00:00:00",
    "updatedBy": "Admin 123",
    "updatedOn": "2020-10-17T00:00:00"
  },
  {
    "startDate": "2020-10-10T12:10:45.741",
    "endDate": "2020-10-11T00:00:00",
    "updatedBy": "Admin 123",
    "updatedOn": "2020-10-12T00:00:00"
  },
  {
    "startDate": "2020-10-01T12:10:45.741",
    "endDate": "2020-10-02T00:00:00",
    "updatedBy": "Admin 123",
    "updatedOn": "2020-10-03T00:00:00"
  },
  {
    "startDate": "2020-10-06T12:10:45.741",
    "endDate": "2020-10-07T00:00:00",
    "updatedBy": "Admin 123",
    "updatedOn": "2020-10-08T00:00:00"
  },
  {
    "startDate": "2020-10-11T12:10:45.741",
    "endDate": "2020-10-12T00:00:00",
    "updatedBy": "Admin 123",
    "updatedOn": "2020-10-13T00:00:00"
  },
  {
    "startDate": "2020-10-16T12:10:45.741",
    "endDate": "2020-10-17T00:00:00",
    "updatedBy": "Admin 123",
    "updatedOn": "2020-10-18T00:00:00"
  },
  {
    "startDate": "2020-10-21T12:10:45.741",
    "endDate": "2020-10-22T00:00:00",
    "updatedBy": "Admin 123",
    "updatedOn": "2020-10-23T00:00:00"
  },
  {
    "startDate": "2020-10-03T12:10:45.741",
    "endDate": "2020-10-06T00:00:00",
    "updatedBy": "Admin 123",
    "updatedOn": "2020-10-09T00:00:00"
  }
]
