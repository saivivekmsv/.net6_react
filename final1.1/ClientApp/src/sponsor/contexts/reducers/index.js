export * from "./app-layout";
export * from "./manage-company";
export * from "./create-plan";
export * from "./manage-eligibility";
export * from "./manage-census";
export * from "./manage-payroll";
export * from "./manage-maintenance";
export * from "./manage-report";
export * from "./manage-mapper";
