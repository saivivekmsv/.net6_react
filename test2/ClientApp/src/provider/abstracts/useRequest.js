import { useState } from "react";
import { isEmpty } from "lodash";
import { useDeepEffect } from "./";

export default function useRequest(request = {}) {
  const [loading, setLoading] = useState(false);
  const [error, setError] = useState(null);
  const [response, setResponse] = useState(request.defaultResponse || null);
  useDeepEffect(
    () => {
      if (!isEmpty(request) && !request.stopTrigger) {
        loadData();
      }
      // eslint-disable-next-line react-hooks/exhaustive-deps
    },
    [request],
    request.triggerOnlyOnUpdate
  );

  const loadData = () => {
    const { method, payload } = request;
    if (isEmpty(request)) {
      return;
    }
    setLoading(true);

    method(payload)
      .then((response) => {
        setResponse(response);
        setError(null);
      })
      .catch((err) => {
        setResponse(request.defaultResponse);
        setError(err);
      })
      .finally(() => {
        setLoading(false);
      });
  };

  return { response, loading, error };
}
