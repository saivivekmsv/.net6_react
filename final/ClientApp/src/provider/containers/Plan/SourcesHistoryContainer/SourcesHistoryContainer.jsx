import React from "react";
import { faTimes } from "@fortawesome/pro-light-svg-icons";
import { ManagePlanLayout, CsplTable as Table } from "../../../components";
import {
  FLOW_TYPES,
  MANAGE_PLAN_ROUTES,
  getPathWithParam,
  usDateFormat,
} from "../../../utils";
import { useRouterParams, useRequest } from "../../../abstracts";
import { getSourcesHistory } from "../../../services";

const columns = [
  {
    label: "Start Date",
    className: "column-start-date",
    columnName: "startDate",
    type: "date",
  },
  {
    label: "End Date",
    className: "column-end-date",
    columnName: "endDate",
    type: "date",
  },
  {
    label: "Update By",
    className: "column-update-by",
    columnName: "updatedBy",
  },
  {
    label: "Update On",
    className: "column-updated-on",
    columnName: "updatedOn",
    type: "date",
  },
];

const ViewMasterSourcesContainer = (props) => {
  const { flow, planId, sourceId } = useRouterParams();
  const { loading, response } = useRequest({
    method: getSourcesHistory,
    defaultResponse: [],
  });

  const buttons = [
    {
      label: "",
      variant: "link",
      type: "button",
      flow: [FLOW_TYPES.EDIT],
      icon: faTimes,
      link: getPathWithParam({
        path: MANAGE_PLAN_ROUTES.MANAGE_SOURCES_NEW,
        pathParam: [flow, planId, sourceId],
      }),
    },
  ];

  return (
    <ManagePlanLayout buttons={buttons} pageFlow={FLOW_TYPES.EDIT}>
      <div className="plan-heading">Sources Effective Date History</div>
      <Table isLoading={loading}>
        <Table.Thead>
          <Table.Tr>
            {columns.map((item, index) => {
              return (
                <Table.Th key={index} className={item.className}>
                  {item.label}
                </Table.Th>
              );
            })}
          </Table.Tr>
        </Table.Thead>
        <Table.Tbody isLoading={loading} totalRecords={response.length}>
          {response.map((history, index) => {
            return (
              <Table.Tr key={index}>
                {columns.map((item, cellIndex) => {
                  return (
                    <Table.Td key={cellIndex} className={item.className}>
                      {item.type === "date"
                        ? usDateFormat(history[item.columnName])
                        : history[item.columnName]}
                    </Table.Td>
                  );
                })}
              </Table.Tr>
            );
          })}
        </Table.Tbody>
      </Table>
    </ManagePlanLayout>
  );
};

export default ViewMasterSourcesContainer;
