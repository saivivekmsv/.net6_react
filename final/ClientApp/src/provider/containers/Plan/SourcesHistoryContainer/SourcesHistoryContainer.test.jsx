import React from "react";
import { render } from "@testing-library/react";

import { BrowserRouter as Router } from "react-router-dom";
import SourcesHistoryContainer from "./SourcesHistoryContainer";
import useRequest from "../../../abstracts/useRequest";
import sourcesEffectiveDateHistory from "../../../mocks/sourcesEffectiveDateHistory.json";
import { CreatePlanProvider } from "../../../contexts";

jest.mock("../../../abstracts/useRequest", () =>
  jest.fn(() => {
    return {
      response: [],
      loading: true,
    };
  })
);

describe("SourcesHistoryContainer - sprint 4", () => {
  test("Should match snapshot with table", () => {
    useRequest.mockReturnValue({
      response: sourcesEffectiveDateHistory,
      loading: false,
    });
    const props = {};

    const wrapper = render(
      <CreatePlanProvider>
        <Router>
          <SourcesHistoryContainer {...props} />
        </Router>
      </CreatePlanProvider>
    );
    const tableBody = wrapper.container.querySelectorAll(".table-body");
    const tableHeadCell = wrapper.container.querySelectorAll(
      ".table-head-cell"
    );
    expect(tableBody.length).toBe(1);
    expect(tableHeadCell.length).toBe(4);
    expect(useRequest).toBeCalled();
    expect(wrapper.baseElement).toMatchSnapshot();
  });
});
