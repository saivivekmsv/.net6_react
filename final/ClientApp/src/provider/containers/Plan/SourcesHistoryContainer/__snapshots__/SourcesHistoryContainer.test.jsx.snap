// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`SourcesHistoryContainer - sprint 4 Should match snapshot with table 1`] = `
<body>
  <div>
    <div
      class="d-flex flex-column manage-company h-100"
    >
      <div
        class="d-flex layout-header align-items-center"
      >
        <div>
          <nav
            aria-label="breadcrumb"
          >
            <ol
              class="breadcrumb"
            >
              <li
                class="breadcrumb-item"
              >
                <a
                  href="#"
                  role="button"
                >
                  <a
                    href="/home"
                  >
                    Home
                  </a>
                </a>
              </li>
              <li
                class="breadcrumb-item"
              >
                <a
                  href="#"
                  role="button"
                >
                  <a
                    href="/plans"
                  >
                    Manage Plan
                  </a>
                </a>
              </li>
            </ol>
          </nav>
          <h5 />
        </div>
        <div
          class="d-flex align-items-center justify-content-center tab-wrapper"
        />
        <div
          class="ml-auto"
        >
          <div>
            <a
              href="/manage-plan/sources-new//"
            >
              <button
                class="inner-layout-actions btn btn-link"
                flow="edit"
                link="/manage-plan/sources-new//"
                type="button"
              >
                <svg
                  aria-hidden="true"
                  class="svg-inline--fa fa-times fa-w-10 "
                  color="#212529"
                  data-icon="times"
                  data-prefix="fal"
                  focusable="false"
                  role="img"
                  style="font-size: 1.5rem;"
                  viewBox="0 0 320 512"
                  xmlns="http://www.w3.org/2000/svg"
                >
                  <path
                    d="M193.94 256L296.5 153.44l21.15-21.15c3.12-3.12 3.12-8.19 0-11.31l-22.63-22.63c-3.12-3.12-8.19-3.12-11.31 0L160 222.06 36.29 98.34c-3.12-3.12-8.19-3.12-11.31 0L2.34 120.97c-3.12 3.12-3.12 8.19 0 11.31L126.06 256 2.34 379.71c-3.12 3.12-3.12 8.19 0 11.31l22.63 22.63c3.12 3.12 8.19 3.12 11.31 0L160 289.94 262.56 392.5l21.15 21.15c3.12 3.12 8.19 3.12 11.31 0l22.63-22.63c3.12-3.12 3.12-8.19 0-11.31L193.94 256z"
                    fill="currentColor"
                  />
                </svg>
              </button>
            </a>
          </div>
        </div>
      </div>
      <div
        class="layout-content h-100 overflow-auto"
      >
        <div
          class="plan-heading"
        >
          Sources Effective Date History
        </div>
        <div
          class="loader-wrapper "
        >
          <div
            class="d-flex flex-column cspl-table undefined"
          >
            <div
              class="d-flex table-head align-items-center undefined"
            >
              <div
                class="d-flex table-row undefined"
              >
                <div
                  class="d-flex table-head-cell column-start-date  undefined"
                >
                  Start Date
                </div>
                <div
                  class="d-flex table-head-cell column-end-date  undefined"
                >
                  End Date
                </div>
                <div
                  class="d-flex table-head-cell column-update-by  undefined"
                >
                  Update By
                </div>
                <div
                  class="d-flex table-head-cell column-updated-on  undefined"
                >
                  Update On
                </div>
              </div>
            </div>
            <div
              class="d-flex flex-column table-body undefined"
            >
              <div
                class="d-flex table-row undefined"
              >
                <div
                  class="d-flex table-cell column-start-date"
                >
                  10/21/2020
                </div>
                <div
                  class="d-flex table-cell column-end-date"
                >
                  10/22/2020
                </div>
                <div
                  class="d-flex table-cell column-update-by"
                >
                  Admin 123
                </div>
                <div
                  class="d-flex table-cell column-updated-on"
                >
                  10/23/2020
                </div>
              </div>
              <div
                class="d-flex table-row undefined"
              >
                <div
                  class="d-flex table-cell column-start-date"
                >
                  10/20/2020
                </div>
                <div
                  class="d-flex table-cell column-end-date"
                >
                  10/21/2020
                </div>
                <div
                  class="d-flex table-cell column-update-by"
                >
                  Admin 123
                </div>
                <div
                  class="d-flex table-cell column-updated-on"
                >
                  10/22/2020
                </div>
              </div>
              <div
                class="d-flex table-row undefined"
              >
                <div
                  class="d-flex table-cell column-start-date"
                >
                  10/15/2020
                </div>
                <div
                  class="d-flex table-cell column-end-date"
                >
                  10/16/2020
                </div>
                <div
                  class="d-flex table-cell column-update-by"
                >
                  Admin 123
                </div>
                <div
                  class="d-flex table-cell column-updated-on"
                >
                  10/17/2020
                </div>
              </div>
              <div
                class="d-flex table-row undefined"
              >
                <div
                  class="d-flex table-cell column-start-date"
                >
                  10/10/2020
                </div>
                <div
                  class="d-flex table-cell column-end-date"
                >
                  10/11/2020
                </div>
                <div
                  class="d-flex table-cell column-update-by"
                >
                  Admin 123
                </div>
                <div
                  class="d-flex table-cell column-updated-on"
                >
                  10/12/2020
                </div>
              </div>
              <div
                class="d-flex table-row undefined"
              >
                <div
                  class="d-flex table-cell column-start-date"
                >
                  10/01/2020
                </div>
                <div
                  class="d-flex table-cell column-end-date"
                >
                  10/02/2020
                </div>
                <div
                  class="d-flex table-cell column-update-by"
                >
                  Admin 123
                </div>
                <div
                  class="d-flex table-cell column-updated-on"
                >
                  10/03/2020
                </div>
              </div>
              <div
                class="d-flex table-row undefined"
              >
                <div
                  class="d-flex table-cell column-start-date"
                >
                  10/06/2020
                </div>
                <div
                  class="d-flex table-cell column-end-date"
                >
                  10/07/2020
                </div>
                <div
                  class="d-flex table-cell column-update-by"
                >
                  Admin 123
                </div>
                <div
                  class="d-flex table-cell column-updated-on"
                >
                  10/08/2020
                </div>
              </div>
              <div
                class="d-flex table-row undefined"
              >
                <div
                  class="d-flex table-cell column-start-date"
                >
                  10/11/2020
                </div>
                <div
                  class="d-flex table-cell column-end-date"
                >
                  10/12/2020
                </div>
                <div
                  class="d-flex table-cell column-update-by"
                >
                  Admin 123
                </div>
                <div
                  class="d-flex table-cell column-updated-on"
                >
                  10/13/2020
                </div>
              </div>
              <div
                class="d-flex table-row undefined"
              >
                <div
                  class="d-flex table-cell column-start-date"
                >
                  10/16/2020
                </div>
                <div
                  class="d-flex table-cell column-end-date"
                >
                  10/17/2020
                </div>
                <div
                  class="d-flex table-cell column-update-by"
                >
                  Admin 123
                </div>
                <div
                  class="d-flex table-cell column-updated-on"
                >
                  10/18/2020
                </div>
              </div>
              <div
                class="d-flex table-row undefined"
              >
                <div
                  class="d-flex table-cell column-start-date"
                >
                  10/21/2020
                </div>
                <div
                  class="d-flex table-cell column-end-date"
                >
                  10/22/2020
                </div>
                <div
                  class="d-flex table-cell column-update-by"
                >
                  Admin 123
                </div>
                <div
                  class="d-flex table-cell column-updated-on"
                >
                  10/23/2020
                </div>
              </div>
              <div
                class="d-flex table-row undefined"
              >
                <div
                  class="d-flex table-cell column-start-date"
                >
                  10/03/2020
                </div>
                <div
                  class="d-flex table-cell column-end-date"
                >
                  10/06/2020
                </div>
                <div
                  class="d-flex table-cell column-update-by"
                >
                  Admin 123
                </div>
                <div
                  class="d-flex table-cell column-updated-on"
                >
                  10/09/2020
                </div>
              </div>
            </div>
          </div>
        </div>
      </div>
    </div>
  </div>
</body>
`;
